gpustack/cmd/prerun.py

```diff
@@ -0,0 +1,303 @@
+        help="Perform pre-run checks and setup s6-overlay configuration for GPUStack.",
+        description="Perform pre-run checks and setup s6-overlay configuration for GPUStack.",
+    )
+    # following args are hidden and used for debugging or advanced usage
+    parser_server.add_argument(
+        "--s6-base-path",
+        type=str,
+        help=argparse.SUPPRESS,
+        default=get_gpustack_env("S6_BASE_PATH"),
+    )
+    parser_server.add_argument(
+        "--skip-data-dir-check",
+        action=OptionalBoolAction,
+        help=argparse.SUPPRESS,
+        default=False,
+    )
+    group = parser_server.add_argument_group("Common settings")
+    group.add_argument(
+        "--config-file",
+        type=str,
+        help="Path to the YAML config file.",
+        default=get_gpustack_env("CONFIG_FILE"),
+    )
+    CommonConfig.set_to_argument_group(group)
+    group = parser_server.add_argument_group("Server settings")
+    ServerConfig.set_to_argument_group(group)
+    group = parser_server.add_argument_group("Worker settings")
+    WorkerConfig.set_to_argument_group(group)
+
+    parser_server.set_defaults(func=run)
+
+
+def run(args: argparse.Namespace):
+    try:
+        cfg = Config.parse_from_args(args)
+        setup_logging(cfg.debug)
+        logger.info(
+            "Starting pre-run checks and setup s6-overlay configuration for GPUStack..."
+        )
+        s6_base_path = args.s6_base_path or "/etc/s6-overlay/s6-rc.d"
+        if args.skip_data_dir_check is False:
+            check_for_data_dir(cfg)
+
+        enabled_services = determine_enabled_services(cfg)
+        dependency_services = determine_dependency_services(cfg)
+        if len(dependency_services) == 0 and len(enabled_services) == 0:
+            logger.info("No extra s6 services for gpustack to enable.")
+        else:
+            logger.info(
+                f"Enabled s6 services: {enabled_services}, dependencies for gpustack: {dependency_services}"
+            )
+        prepare_s6_overlay(enabled_services, dependency_services, s6_base_path)
+
+        check_ports_availability(cfg, *enabled_services)
+        if "postgres" in enabled_services:
+            prepare_postgres_config(cfg)
+        if cfg.gateway_mode == GatewayModeEnum.embedded:
+            prepare_gateway_config(cfg)
+        logger.info("Pre-run checks and setup completed successfully.")
+    except Exception as e:
+        logger.fatal(f"Failed to pre-check the configuration: {e}")
+
+
+def check_for_data_dir(cfg: Config):
+    if cfg.data_dir != Config.get_data_dir():
+        raise Exception(
+            f"Custom data directory '{cfg.data_dir}' is not supported in container environment."
+        )
+
+
+def ports_for_services(cfg: Config) -> Dict[int, str]:
+    ports = {}
+    # postgres
+    if not cfg.database_url and cfg.server_role() in [
+        Config.ServerRole.SERVER,
+        Config.ServerRole.BOTH,
+    ]:
+        ports[cfg.database_port] = "postgres"
```

Comment:
replace with the postgres_services

Reply:
Should the service in s6 also be renamed?